scripts/pre_commit_linter.py

```diff
@@ -1061,6 +1071,65 @@ def _check_docstrings(all_files):
                         print '%s --> Line %s: %s' % (
                             filename, line_num + 1, multiline_docstring_message)
 
+    # Check that the args in the docstring are listed in the same
+    # order as they appear in the function definition.
+    def _check_docstrings_arg_order(filename):
+        file_result = True
+        with open(filename, 'r') as f:
+            ast_file = ast.walk(ast.parse(f.read()))
+            func_defs = [n for n in ast_file if isinstance(n, ast.FunctionDef)]
+            for func in func_defs:
+
+                def get_args_list_from_function_definition(function_node):
+                    return [a.id for a in function_node.args.args]
+
+                def parse_arg_list_from_docstring(docstring):
+                    # Note: this function returns everything after the 'Args:'
+                    # title within the docstring, including 'Returns:' etc if
+                    # they exist.
+                    if docstring is not None and 'Args:' in docstring:
+                        docstring_post_args_title = docstring.split('Args:')[1]
+                        ds_arg_lines = docstring_post_args_title.split('\n')
+                        return [line for line in ds_arg_lines if line != '']
+                    else:
+                        return []
+
+                func_def_args = get_args_list_from_function_definition(func)
+                docstring = ast.get_docstring(func)
+                docstring_args = parse_arg_list_from_docstring(docstring)
+
+                def check_arg_order(func_name, func_def_args, docstring_args):
+                    num_args_in_docstring = len(docstring_args)
+                    if num_args_in_docstring > 0:
+                        for i, arg_name in enumerate(func_def_args):
+                            if i < len(docstring_args):
+                                if func_def_args[i] not in docstring_args[i]:
```

Comment:
Does this correctly handle the case where an arg description spans more than a single line?

Reply:
In fact, if you look at https://travis-ci.org/oppia/oppia/jobs/434325692, it seems that the lint check passes but there are a lot of docstring-related error messages. I think this suggests that the lint check might not be working as intended...? (In general the lint check should block submission if the criteria are not met.)